mesonbuild/linkers/detect.py

```diff
@@ -71,7 +72,10 @@ def guess_win_linker(env: 'Environment', compiler: T.List[str], comp_class: T.Ty
         elif not invoked_directly:
             return linkers.ClangClDynamicLinker(
                 for_machine, override, exelist=compiler, prefix=comp_class.LINKER_PREFIX,
-                version=search_version(o), direct=False, machine=None)
+                version=search_version(o), direct=False, machine=None,
+                rsp_syntax=RSPFileSyntax.MSVC
+                if (invoked_directly or comp_class.get_argument_syntax() == 'msvc')
```

Comment:
You are inside a `if not invoked_directly`. Therefore, you don't need to test for `invoked_directly` here...

Reply:
done